server/sonar-server/src/main/java/org/sonar/server/platform/monitoring/SonarQubeMonitor.java

```diff
@@ -111,6 +141,44 @@ public String name() {
     attributes.put("Logs Dir", settings.getString(ProcessProperties.PATH_LOGS));
     attributes.put("Logs Level", getLogLevel());
     return attributes;
+  }
+
+  private static void addIfNotNull(String key, @Nullable String value, Map<String, Object> attributes) {
+    if (value != null) {
+      attributes.put(key, value);
+    }
+  }
+
+  private static void addIfNotEmpty(String key, List<String> values, Map<String, Object> attributes) {
+    if (!values.isEmpty()) {
+      attributes.put(key, COMMA_JOINER.join(values));
+    }
+  }
+
+  private enum IdentityProviderToName implements Function<IdentityProvider, String> {
+    INSTANCE;
+
+    @Override
+    public String apply(@Nonnull IdentityProvider input) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this "Nonnull" annotation to honor the overridden method's contract. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS2638)